src/stackwalk.c

```diff
@@ -1108,6 +1108,42 @@ JL_DLLEXPORT void jl_print_backtrace(void) JL_NOTSAFEPOINT
     jlbacktrace();
 }
 
+JL_DLLEXPORT void jl_print_task_backtraces(void)
+{
+    jl_task_t *t;
+    jl_ptls_t ptls;
+    arraylist_t *live_tasks;
+    size_t i, j, n;
+
+    for (i = 0; i < jl_n_threads; i++) {
+        ptls = jl_all_tls_states[i];
+        live_tasks = &ptls->heap.live_tasks;
+        n = live_tasks->len;
+        jl_safe_printf("==== Thread %d (%zu live tasks)\n",
+                ptls->tid + 1, n + 1);
+        t = ptls->root_task;
+        jl_safe_printf("     ---- Root task (%p)\n", t);
+        jl_safe_printf("          (sticky: %d, started: %d, state: %d, tid: %d)\n",
+                t->sticky, t->started, jl_atomic_load_relaxed(&t->_state),
+                jl_atomic_load_relaxed(&t->tid) + 1);
+        jlbacktracet(t);
+
+        void **lst = live_tasks->items;
+        for (j = 0; j < live_tasks->len; j++) {
+            t = (jl_task_t *)lst[j];
+            jl_safe_printf("     ---- Task %lu (%p)\n", j + 1, t);
+            jl_safe_printf("          (sticky: %d, started: %d, state: %d, tid: %d)\n",
+                    t->sticky, t->started, jl_atomic_load_relaxed(&t->_state),
+                    jl_atomic_load_relaxed(&t->tid) + 1);
+            if (t->stkbuf != NULL)
+                jlbacktracet(t);
+            else
+                jl_safe_printf("      no stack\n");
+        }
+    }
+    jl_safe_printf("==== Done\n");
```

Comment:
```suggestion
    jl_safe_printf("==== End of Thread %d\n", ptls->tid + 1);
```
🤷 i think this might make the log easier to read if there's a longgggg gap between the start and end?